File: api/src/main/java/net/onelitefeather/pandorascluster/api/utils/NumberUtil.java
package net.onelitefeather.pandorascluster.api.utils;

public final class NumberUtil {

    private NumberUtil() {
        throw new IllegalStateException("Utility class");
    }

    public static int floor(Double number) {
        final int floor = number.intValue();
        return floor == number ? floor : floor - (int) (Double.doubleToRawLongBits(number) >>> 63);
    }

    public static int locToBlock(double loc) {
        return floor(loc);

    }
}

File: api/src/main/kotlin/net/onelitefeather/pandorascluster/api/utils/NumberUtil.kt
package net.onelitefeather.pandorascluster.api.utils

interface NumberUtil {

    fun floor(num: Double): Int {
        val floor = num.toInt()
        return if (floor.toDouble() == num) floor else floor - (java.lang.Double.doubleToRawLongBits(num) ushr 63).toInt()
    }

    fun locToBlock(loc: Double): Int {
        return floor(loc)
    }
}
